diff -u -p a/drivers/scsi/dpt_i2o.c b/drivers/scsi/dpt_i2o.c
--- a/drivers/scsi/dpt_i2o.c
+++ b/drivers/scsi/dpt_i2o.c
@@ -1001,7 +1001,6 @@ static int adpt_install_hba(struct scsi_

        // Initializing the spinlocks
        spin_lock_init(&pHba->state_lock);
-       spin_lock_init(&adpt_post_wait_lock);

        if(raptorFlag == 0){
                printk(KERN_INFO "Adaptec I2O RAID controller"
