diff -u -p a/arch/ia64/kernel/mca.c b/arch/ia64/kernel/mca.c
--- a/arch/ia64/kernel/mca.c
+++ b/arch/ia64/kernel/mca.c
@@ -290,7 +290,6 @@ static void ia64_mlogbuf_finish(int wait
 {
        BREAK_LOGLEVEL(console_loglevel);

-       spin_lock_init(&mlogbuf_rlock);
        ia64_mlogbuf_dump();
        printk(KERN_EMERG "mlogbuf_finish: printing switched to urgent mode, "
                "MCA/INIT might be dodgy or fail.\n");
